app/src/test/java/org/simple/clinic/addidtopatient/searchforpatient/AddIdToPatientSearchScreenControllerTest.kt

```diff
@@ -0,0 +1,79 @@
+package org.simple.clinic.addidtopatient.searchforpatient
+
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.times
+import com.nhaarman.mockito_kotlin.verify
+import io.reactivex.subjects.PublishSubject
+import junitparams.JUnitParamsRunner
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.util.RxErrorsRule
+import org.simple.clinic.widgets.UiEvent
+
+@RunWith(JUnitParamsRunner::class)
+class AddIdToPatientSearchScreenControllerTest {
+
+  @get:Rule
+  val rxErrorsRule = RxErrorsRule()
+
+  private val screen: AddIdToPatientSearchScreen = mock()
+
+  private lateinit var controller: AddIdToPatientSearchScreenController
+  private val uiEvents = PublishSubject.create<UiEvent>()
+
+  @Before
+  fun setUp() {
+    controller = AddIdToPatientSearchScreenController()
+
+    uiEvents.compose(controller).subscribe { uiChange -> uiChange(screen) }
+  }
+
+  @Test
+  fun `search button should remain enabled only when name is present`() {
+    uiEvents.onNext(SearchQueryNameChanged(""))
+    uiEvents.onNext(SearchQueryNameChanged("foo"))
+    uiEvents.onNext(SearchQueryNameChanged(" "))
+    uiEvents.onNext(SearchQueryNameChanged("bar"))
+
+    verify(screen, times(2)).showSearchButtonAsDisabled()
+    verify(screen, times(2)).showSearchButtonAsEnabled()
+  }
+
+  @Test
+  fun `when search is clicked with empty name then a validation error should be shown`() {
+    uiEvents.onNext(SearchQueryNameChanged(""))
+    uiEvents.onNext(SearchClicked())
+
+    verify(screen, times(1)).setEmptyFullNameErrorVisible(true)
```

Comment:
Do you want to explicitly say `times(1)` here? It's not an issue if you want to keep it.

Reply:
It's redundant. Removed.